namerd/iface/control-http/src/main/scala/io/buoyant/namerd/iface/HttpControlService.scala

```diff
@@ -70,47 +70,92 @@ object HttpControlService {
     val prefix = "/api/1/dtabs"
     val prefixSlash = s"$prefix/"
 
-    def unapply(path: String): Option[Option[String]] =
-      if (path == prefix) Some(None)
-      else if (!path.startsWith(prefixSlash)) None
-      else path.stripPrefix(prefixSlash) match {
-        case "" => Some(None)
-        case ns => Some(Some(ns))
+    def unapply(request: Request): Option[(Method, Option[Ns])] =
+      if (request.path == prefix) Some(request.method, None)
+      else if (!request.path.startsWith(prefixSlash)) None
+      else request.path.stripPrefix(prefixSlash) match {
+        case "" => Some(request.method, None)
+        case ns => Some(request.method, Some(ns))
       }
   }
 
+  case class InvalidPathException(path: String, underlying: Exception)
+    extends Exception(s"Invalid path: $path\n${underlying.getMessage}", underlying)
+
+  trait NsPathUri {
+    val prefix: String
+
+    def unapply(request: Request): Option[(Ns, Path)] = {
+      if (request.path.startsWith(prefix)) {
+        val ns = request.path.stripPrefix(prefix)
+        val path = try {
+          Path.read(request.getParam("path"))
+        } catch {
+          case e: IllegalArgumentException =>
+            throw InvalidPathException(request.getParam("path"), e)
+        }
+        Some(ns, path)
+      } else {
+        None
+      }
+    }
+  }
+
+  object BindUri extends NsPathUri {
+    val prefix = "/api/1/bind"
```

Comment:
tioli, tiny: might be good to put `"/api/1"` as a base API URL in `NsPathUri` and construct this and friends via string interpolation, rather than have it repeated for each prefix.